routes_test.go

```diff
@@ -621,3 +621,24 @@ func TestRouteContextHoldsFullPath(t *testing.T) {
 	w := performRequest(router, http.MethodGet, "/not-found")
 	assert.Equal(t, http.StatusNotFound, w.Code)
 }
+
+// Reproduction test for the bug of issue #2843
+func TestRouteTrailingSlashRoute(t *testing.T) {
```

Comment:
Perhaps we don't need a new function, could we integrate it into the function 'TestRouterNotFound'?
https://github.com/gin-gonic/gin/blob/4e7584175d7f2b4245249e769110fd1df0d779db/routes_test.go#L434-L470